pkg/storage/db.go

```diff
@@ -320,11 +320,17 @@ func (db *DB) GetPartitions(ctx context.Context) ([]domain.Partition, error) {
 			End:       end,
 		})
 	}
-
 	if err := rows.Close(); err != nil {
 		return nil, err
 	}
-
+	for _, v := range partitions {
+		row := db.sqldb.QueryRow("SELECT count(*) FROM " + v.Name)
+		var count int
+		if err := row.Scan(&count); err != nil {
+			return nil, err
+		}
+		v.Count = count
```

Comment:
task: Unfortunately, this doesn't work the way you expect it to in Go. I put together a playground to demonstrate why: https://play.golang.org/p/xk9C3ARk3B1. There is also a good write-up about this at https://dave.cheney.net/2017/04/29/there-is-no-pass-by-reference-in-go.

tl;dr You need adjust the loop to look like:

```golang
for offset := range partitions {
  // <query for the count>
  partitions[offset].Count = count
}
```

I recommend writing or modifying one of the unit tests in db_test.go in order to verify that the we get the correct behavior from setting the counts.

Reply:
Fixed that. Thank you